### PR TITLE
Remove unnecessary last quote buttons

### PR DESCRIPTION
**Pull request by bevekashaivelt**
Some posts don't have quotes, so it is not necessary to have a button to quote only last. This fix checks if the post has a quote, and only then does it add the extra button.

### CODE REVIEW

**Comment by bevekashaivelt** [comment]
Last quote only didn't work on the posting page. This last commit fixes it. (

File: extension/js/removeNestedQuotes.js [hunk 1]
```diff
@@ -33,3 +33,62 @@ if (onlyLastQuote) {
...
+
+	// Get text selection - not only the post content :(
+	// IE9 must use the document.selection method but has the *.getSelection so we just force no IE
+	if (window.getSelection && !is_ie && !window.opera) {
+		theSelection = window.getSelection().toString();
+	} else if (document.getSelection && !is_ie) {
+		theSelection = document.getSelection();
+	} else if (document.selection) {
+		theSelection = document.selection.createRange().text;
+	}
```
Comment (ShimonLeb) [inline extension/js/removeNestedQuotes.js:62]: no need to check for IE as this is a chrome extension

File: extension/js/removeNestedQuotes.js [hunk 2]
```diff
@@ -33,3 +33,62 @@ if (onlyLastQuote) {
...
+	var i;
+
+	if (l_wrote === undefined) {
+		// Backwards compatibility
+		l_wrote = 'wrote';
+	}
+
+	if (document.all) {
+		divarea = document.all[message_name];
+	} else {
```
Comment (ShimonLeb) [inline extension/js/removeNestedQuotes.js:50]: why do you need this? why not just use `getElementById`? 
`document.all` is Deprecated

**Comment by ShimonLeb** [comment]
as per @mordechairoth requested please use all camelCase

**Comment by mayim-chayim** [comment]
as per @mordechairoth requested please use all camelCase

File: extension/js/removeNestedQuotes.js (hunk 1 above)
Comment (mayim-chayim) [inline extension/js/removeNestedQuotes.js:62]: why do you need this? why not just use `getElementById`?
`document.all` is Deprecated

File: extension/js/removeNestedQuotes.js (hunk 2 above)
Comment (mayim-chayim) [inline extension/js/removeNestedQuotes.js:50]: why do you need this? why not just use `getElementById`?
`document.all` is Deprecated

**Comment by bevekashaivelt** [comment]
I copy/pasted the code from phpBB addQuote method adding only the removeNestedQuotes call. I wanted to make sure it works, and theirs works.

File: extension/js/removeNestedQuotes.js [hunk 3]
```diff
@@ -33,3 +33,62 @@ if (onlyLastQuote) {
...
+			insert_text(username + ' ' + l_wrote + ':' + '\n');
+			var lines = split_lines(theSelection);
+			for (i = 0; i < lines.length; i++) {
+				insert_text('> ' + lines[i] + '\n');
+			}
+		}
+	}
+
+	return;
+}
```
Comment (mayim-chayim) [inline extension/js/removeNestedQuotes.js:94]: This is all code that is needed, there is no point to use the last quote when using text selection or when bbcode is disabled 

```suggestion
function lastaddquote(post_id, username) {
	let messageName= 'message_' + post_id;
	let theSelection = '';
	let divArea = false;

	divArea = document.getElementById(messageName);

	if (divArea.innerHTML) {
		theSelection = divArea.innerHTML.replace(/<br>/ig, '\n');
		theSelection = theSelection.replace(/<br\/>/ig, '\n');
		theSelection = theSelection.replace(/&lt\;/ig, '<');
		theSelection = theSelection.replace(/&gt\;/ig, '>');
		theSelection = theSelection.replace(/&amp\;/ig, '&');
		theSelection = theSelection.replace(/&nbsp\;/ig, ' ');
	}  else if (divArea.textContent) {
		theSelection = divArea.textContent;
	} else if (divArea.firstChild.nodeValue) {
		theSelection = divArea.firstChild.nodeValue;
	}

	if (theSelection) {
		let text = '[quote="' + username + '"]' + theSelection + '[/quote]'
		insert_text(removeNestedQuotes(text));
	}

	return;
}
```

**Pull request merged**